### ISSUE TITLE
[enhancement] warning/incomplete support?

### ISSUE THREAD

**Issue by thehesiod** [DataDog/dd-trace-py#414]
So right now there are two types of traces, regular traces and "error" traces.  I have a scenario where I speculatively issue several requests in parallel and take the fastest one to complete...cancelling the other asyncio tasks.  As "cancelling" throws an canceled exception, these end up getting marked as errors....however they're not really errors, and they not non-errors.  I also don't want to filter them out because I want to see how much resources they used.  I'm wondering if there could be a way to mark them like "warning" or something.  Maybe greyed out or something.

This is a "nice to have" request.

Thoughts?

**Comment by clutchski** [comment]
@thehesiod can you include a little code snippet of how you're cancelling the traces? I think an easy work around would be to add a func like:

```
def _mark_span_cancelled(span):
       span.error = 0
       span.set_tag(errors.ERROR_MSG, exc_val)
       span.set_tag(errors.ERROR_TYPE, exc_type_str)
       span.set_tag(errors.ERROR_STACK, tb)
       span.set_tag("async.cancelled_or_discarded", "true")
```

and then call than when you can cancel. We could add a `span.clear_error()` helper at some point as well.

**Comment by thehesiod** [comment]
thanks, ya it's pretty easy to not mark the trace as having been cancelled, however it would be nice to show in the UI that these traces as "incomplete"

basically I have something like this:
```python
def relay_wrapper(func: asyncio.coroutine, num_parallel: int):
    """
    Will wrap `func` and run `num_parallel` in parallel, returning result from first
    that completes, cancelling others.

    :param func: async method to call
    :param num_parallel: number of calls to run in parallel
    :return:
    """
    async def wrapper(*args, **kwargs):
        done, pending = await asyncio.wait([func(*args, **kwargs) for _ in range(num_parallel)], return_when=asyncio.FIRST_COMPLETED)
        for task in pending:
            await cancel_if_running(task)

        # find first non-exception result
        for task in done:
            if not task.exception():
                return task.result()

        # otherwise return first result
        return done.pop().result()

    return wrapper
```

and use it like this, suppose `get_object_` is equivalent to botocore's s3 client's `get_object`
```python
self.get_object = relay_wrapper(self.get_object, num_parallel_gets)
```

This way the first one "wins", and the others are cancelled.  This helped reduce our latency tremendously...however ends up showing a ton of error spans.  If I mark them as success with the above I'd have to dig through each one to see which one was relevant.  Kinda would like a greylist of spans maybe?  I could then even mark the ones that completed but I did not use (more than one can complete with `FIRST_COMPLETED`) grey.

**Comment by brettlangdon** [comment]
@thehesiod I know this issue has been open for a while, are you still having trouble here?

**Comment by thehesiod** [comment]
this isn't trouble, it just a feature request, like being able mark traces as "warning", instead of normal or error.  No biggie.

**Comment by inikolaev** [comment]
I have somewhat similar need to be able to see `4xx` responses in the graph, without manually filtering all requests for those.

**Comment by techdragon** [comment]
I have a similar issue where I get spurious boto3 library call errors in code that's performing checks to ensure objects are retrievable from s3. These are handled errors but the tracing propagates them as errors into the datadog UI

**Comment by mcreenan** [comment]
> I have a similar issue where I get spurious boto3 library call errors in code that's performing checks to ensure objects are retrievable from s3. These are handled errors but the tracing propagates them as errors into the datadog UI

I just ran into the same issue while using the boto3 cognito-idp client.  The `admin_get_user` call raises a `UserNotFoundException` if there is no user. Even if you handle that exception, it gets logged as an error in APM. Unfortunately there is no other way, that I can tell, to check if a user exists in cognito.
